src/rpc/connection/crypto.c

```diff
@@ -0,0 +1,275 @@
+#include <stdint.h>
+#include "sb-common.h"
+#include "rpc/sb-rpc.h"
+#include "tweetnacl.h"
+
+/* server security */
+unsigned char serverlongtermsk[32];
+unsigned char servershorttermpk[32];
+unsigned char servershorttermsk[32];
+
+/* client security */
+unsigned char clientshorttermpk[32];
+
+/* shared secrets */
+unsigned char clientshortserverlong[32];
```

Comment:
i don't agree. there can be more than on shared secret (clientshortservershort, clientshortserverlong), so naming it `sharedsecret` could confuse the reader.